src/main/python/deep_qa/layers/backend/masked_batch_dot.py

```diff
@@ -0,0 +1,45 @@
+from keras.layers import Layer
+from overrides import overrides
+
+from ...tensors.masked_operations import masked_batch_dot
+
+
+class MaskedBatchDot(Layer):
+    """
+    This ``Layer`` calls ``masked_batch_dot()`` on two inputs ``tensor_a`` and ``tensor_b``.
+    This function will work for tensors of arbitrary size as long as
+    ``abs(K.ndim(tensor_a) - K.ndim(tensor_b)) < 1``, due to limitations in ``K.batch_dot()``.
+
+    We always assume the dimension to perform the dot is the last one, and that the
+    masks have one fewer dimension that the tensors.
+
+    Inputs:
+        - tensor_a: tensor with ``ndim >= 2``.
+        - tensor_b: tensor with ``ndim >= 2``.
+
+    Output:
+        - a_dot_b
+    """
+    def __init__(self, **kwargs):
+        self.supports_masking = True
+        super(MaskedBatchDot, self).__init__(**kwargs)
+
+    @overrides
+    def compute_mask(self, inputs, mask=None):
+        # pylint: disable=unused-argument
+        # We do not need a mask beyond this layer.
```

Comment:
The fact that you're not returning a mask should be in the class docstring, including some kind of statement saying that `MaskedBatchDot` returns zeros in places that were masked.  If this then gets fed into a softmax, for instance, your computation will be incorrect.